Update readme: Removed v1.7 from supported releases

shazib-summar opened the pull request:
According to the [release cycle documentation](https://crossplane.io/docs/master/reference/release-cycle.html#release-cycle)

This results in four releases per year, with the most recent three releases being maintained at any given time. _**When a new release is cut, the fourth most recent release reaches end of life (EOL)**_. Users can expect any given release to be maintained for nine months.

<!--
Thank you for helping to improve Crossplane!

Please read through https://git.io/fj2m9 if this is your first time opening a
Crossplane pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes
Removed v1.7 from the table of supported releases as v1.10 was released a week ago, making v1.7 EOL'd.
<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open Crossplane issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

-->

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->
I did not test any code as the changes are only in the readme.


[contribution process]: https://git.io/fj2m9

hasheddan commented on README.md:
```diff
@@ -19,7 +19,6 @@ documentation].
 
 | Release | Release Date |      EOL      |
 |:-------:|:------------:|:-------------:|
-|   v1.7  | Mar 22, 2022 |    Oct 2022   |
```
I believe per the preceding paragraph we should also add the upcoming release (v1.11) to the table. I believe we would tentatively be targeting January 17th with EOL Oct 2023.

hasheddan reviewed (commented):
Thanks @shazib-summar! One comment and we’ll also need DCO sign off here

shazib-summar commented:
Really sorry @hasheddan, but I screwed up the branch because I didnt know I had to sign my commits. I have addressed your comments in #3411. Closing this PR.
